### PR TITLE
Enable SMB logging for the Community Edition

### PR DESCRIPTION
**Pull request by mmattel**
## Description
New config.php parameter ``smb.logging.enable`` to enable SMB logging.
A parameter for this already exists for the Enterprise edition ``wnd.logging.enable`` , but to seperate this properly, a community parameter is needed. The code for this already existed, it was just about to add the new parameter additionally and document it.

## Motivation and Context
Discussion with @PVince81 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

https://github.com/owncloud/documentation/issues/3715

### CODE REVIEW

**Comment by PVince81** [comment]
I'm fine with this if you guys are too @jvillafanez @butonic

**Comment by jvillafanez** [comment]
I'd just rename the configuration option: less changes => less problems. Having both seems useless to me.
That option is merely for debugging, so there shouldn't be anyone actively using it, specially due to its heavy verbosity. Changing the name doesn't have any impact.

File: apps/files_external/lib/Lib/Storage/SMB.php [hunk 1]
```diff
@@ -659,50 +659,70 @@ public function test() {
 	}
 
 	/**
-	 * @param string $message
-	 * @param int $level
+	 * Return either 'wnd' or 'smb' if the related config values has been enabled
+	 * Return a an existing string if set by the caller
```
Comment (phil-davis) [inline apps/files_external/lib/Lib/Storage/SMB.php]: s/a an existing/an existing

Comment (mmattel) [reply apps/files_external/lib/Lib/Storage/SMB.php]: oh, will correct this. Thought that a return without an preceeding @ is ok to phpdoc not to parse it as a keyword.

**Comment by mmattel** [comment]
@jvillafanez ``wnd`` is exclusively used by the Enterprise Edition (see https://marketplace.owncloud.com/apps/windows_network_drive) and it is already documented in documentation/admin_manual/configuration/server/config_sample_php_parameters.rst (This enables debug logs for the windows_network_drive app.)

The intention was to not mix up Enterprise and Community parameters and keep this clearly seperated...

**Comment by jvillafanez** [comment]
That part of the code is already complex enough to add even more complexity, that's why if the easiest solution (renaming the variable) works, then let's go with it.

Renaming the variable doesn't have any impact in any of both apps (SMB and WND).

The fact that "wnd" is there is because of a slip when this was added here, and kind of laziness to not change it afterwards (me included). There is no need to keep it if we're going to change it. If WND wanted to add special logs or anything, it has to do it on its own app anyway.

We'll have to update the documentation accordingly, but other than that I don't think we need any degree of compatibility here: that switch isn't intended to be actively used so it shouldn't matter if we change the switch.

**Comment by mmattel** [comment]
Just to have the tasks clear:
- reject the current code changes in this PR
- rename the parameter ``wnd.logging.enable`` to ``smb.logging.enable`` and ``wnd`` to ``smb``
- keep the config.sample.php documentation part
- change in an seperate PR the current documentation (see above) to reflect this change

Let me know if this is correct or if I missed something.

**Comment by jvillafanez** [comment]
Yes, that's it

**Review by jvillafanez — APPROVED** [review]

**Comment by mmattel** [comment]
If this gets merged, I will do a backport to stable10 if wanted.

**Comment by mmattel** [comment]
Ready to merge?

**Comment by mmattel** [comment]
@PVince81 
Shall I backport?

**Comment by PVince81** [comment]
@mmattel yes, please go ahead. Thanks

**Comment by phil-davis** [comment]
Backport stable10 #30244

**Pull request merged**